app/controllers/crypto_address_for_channels_controller.rb

```diff
@@ -0,0 +1,97 @@
+require "eth"
+require "ed25519"
+require "base58"
+
+class CryptoAddressForChannelsController < ApplicationController
+  include Eth
+  before_action :authenticate_publisher!
+
+  def index
+    @crypto_addresses_for_channel = CryptoAddressForChannel.where(channel_id: params[:channel_id])
+    render(json: @crypto_addresses_for_channel)
+  end
+
+  def create
+    signature = params[:transaction_signature]
+    account_address = params[:account_address]
+    chain = params[:chain]
+    message = params[:message]
+    channel = Channel.find(params[:channel_id])
+
+    # check to make sure the user owns the address
+    verified = case chain
+    when "SOL"
+      verify_solana_address(signature, account_address, message)
+    when "ETH"
+      verify_ethereum_address(signature, account_address, message)
+    else
+      false
+    end
+
+    # Create new crypto address, and remove any other addresses on the same chain for the channel
+    success = verified && replace_crypto_address_for_channel(account_address, chain, channel)
+
+    respond_to do |format|
+      format.json {
+        if success
+          render(json: {crypto_address_for_channel: success}, status: 201)
+        else
+          render(json: {errors: "address could not be verified"}, status: 400)
+        end
+      }
+    end
+  end
+
+  def change_address
+    account_address = params[:address]
+    chain = params[:chain]
+    channel = Channel.find(params[:channel_id])
+
+    success = replace_crypto_address_for_channel(account_address, chain, channel)
+
+    respond_to do |format|
+      format.json {
+        if success
+          render(json: {crypto_address_for_channel: success}, status: 200)
+        else
+          render(json: {errors: "address could not be updated"}, status: 400)
+        end
+      }
+    end
+  end
+
+  def verify_solana_address(signature, address, message)
```

Comment:
If you want an example of how we're handling this for brave-talk here's some good code to take a look at for the backend service: https://github.com/brave-intl/subscriptions/pull/312/files#diff-8593b22881785b8c7084d5f77364bcf5216445c2e32306da1513056e5d6b44e0R205-R246